presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/OptimizeMixedDistinctAggregations.java

```diff
@@ -0,0 +1,507 @@
+        groupByKeys.add(groupSymbol);
+
+        // This map has mapping only for aggregation on non-distinct symbols which the new AggregationNode handles
+        ImmutableMap.Builder aggregationOutputSymbolsMapBuilder = ImmutableMap.builder();
+        AggregationNode nonDistinctAggregation = createNonDistinctAggregation(
+                aggregateInfo,
+                distinctSymbol,
+                duplicatedDistinctSymbol,
+                groupByKeys,
+                groupIdNode,
+                markDistinctNode,
+                aggregationOutputSymbolsMapBuilder,
+                symbolAllocator,
+                idAllocator);
+
+        // 3. Add new project node that adds IF expressions
+        return createProjectNode(
+                nonDistinctAggregation,
+                aggregateInfo,
+                distinctSymbol,
+                groupSymbol,
+                groupBySymbols,
+                aggregationOutputSymbolsMapBuilder.build(),
+                symbolAllocator,
+                idAllocator);
+    }
+
+    // Returns false if either mask symbol or any of the symbols in aggregations is not comparable
+    private static boolean checkAllEquatableTypes(AggregateInfo aggregateInfo, SymbolAllocator symbolAllocator)
+    {
+        for (Symbol symbol : aggregateInfo.getOriginalNonDistinctAggregateArgs()) {
+            Type type = symbolAllocator.getTypes().get(symbol);
+            if (!type.isComparable()) {
+                return false;
+            }
+        }
+
+        return symbolAllocator.getTypes().get(aggregateInfo.getMask()).isComparable();
+    }
+
+    private static GroupIdNode createGroupIdNode(
+            List<Symbol> groupBySymbols,
+            List<Symbol> nonDistinctAggregateSymbols,
+            Symbol distinctSymbol,
+            Symbol duplicatedDistinctSymbol,
+            Symbol groupSymbol,
+            Set<Symbol> allSymbols,
+            PlanNode source,
+            PlanNodeIdAllocator idAllocator)
+    {
+        List<List<Symbol>> groups = new ArrayList<>();
+        // g0 = {group-by symbols + allNonDistinctAggregateSymbols}
+        // g1 = {group-by symbols + Distinct Symbol}
+        // symbols present in Group_i will be set, rest will be Null
+
+        //g0
+        Set<Symbol> group0 = new HashSet<>();
+        group0.addAll(groupBySymbols);
+        group0.addAll(nonDistinctAggregateSymbols);
+        groups.add(ImmutableList.copyOf(group0));
+
+        // g1
+        Set<Symbol> group1 = new HashSet<>();
+        group1.addAll(groupBySymbols);
+        group1.add(distinctSymbol);
+        groups.add(ImmutableList.copyOf(group1));
+
+        return new GroupIdNode(
+                idAllocator.getNextId(),
+                source,
+                groups,
+                allSymbols.stream().distinct().collect(Collectors.toMap(
+                        symbol -> symbol,
+                        symbol -> (symbol.equals(duplicatedDistinctSymbol) ? distinctSymbol : symbol))),
```

Comment:
it is added in line 199 and 204. The newSymbol(i.e- duplicatedDistinctSymbol) is added to allSymbols. This is to handle cases where both distinct and non-distinct agg are on the same column.